staging/src/k8s.io/kubectl/pkg/cmd/logs/logs_nodes.go

```diff
@@ -0,0 +1,584 @@
+		This command retrieves logs for the node. The default mode is to query the
+		systemd journal or WinEvent entries on supported operating systems,
+		which allows searching, time based filtering, and service based filtering.
+		You may also use the --path argument to see a list of log files available
+		under /var/logs/ and view those contents directly.
+
+		Node logs may contain sensitive output and so are limited to privileged node
+		administrators. The system:node-admins role grants this permission by default.`))
+
+	logsNodesExample = templates.Examples(i18n.T(`
+		# Show kubelet and crio journald logs from all masters
+		kubectl logs nodes --role master -u kubelet -u crio
+
+		# Show kubelet log file (/var/log/kubelet/kubelet.log) from all Windows worker nodes
+		kubectl logs nodes --label kubernetes.io/os=windows -u kubelet
+
+		# Display docker WinEvent logs from a specific Windows worker node
+		kubectl logs nodes <node-name> --service docker
+
+		# Show crio journald logs from a specific Linux node
+		kubectl logs nodes <node-name> --service crio`))
+
+	nodeSelectorTail int64 = 10
+)
+
+type NodeLogsOptions struct {
+	Resources []string
+	Selector  string
+	Role      string
+	Options   runtime.Object
+
+	// the log path to fetch
+	Path string
+	// --path=journal specific arguments
+	Pattern              string
+	PatternCaseSensitive bool
+	Boot                 int64
+	BootChanged          bool
+	Services             []string
+	SinceTime            string
+	UntilTime            string
+	TailLines            int64
+	Output               string
+	// output format arguments
+	// raw is set to true when we are viewing the journal and wish to skip prefixing
+	Raw    bool
+	Unify  bool
+	Node   string
+	Prefix bool
+
+	InsecureSkipTLSVerifyBackend bool
+
+	ConsumeRequestFn func(rest.ResponseWrapper, io.Writer) error
+
+	Object           runtime.Object
+	GetNodeTimeout   time.Duration
+	RESTClientGetter genericclioptions.RESTClientGetter
+	LogsForObject    polymorphichelpers.LogsForObjectFunc
+
+	genericclioptions.IOStreams
+
+	TailSpecified bool
+}
+
+func NewNodeLogsOptions(streams genericclioptions.IOStreams) *NodeLogsOptions {
+	return &NodeLogsOptions{
+		Path:                 "journal",
+		IOStreams:            streams,
+		PatternCaseSensitive: true,
+	}
+}
+
+// NewCmdNodeLogs creates a new logs command that supports OpenShift resources.
+func NewCmdNodeLogs(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewNodeLogsOptions(streams)
+	cmd := &cobra.Command{
+		Use:                   logsNodesUsageStr,
+		DisableFlagsInUseLine: true,
+		Short:                 "Display and filter node logs",
+		Long:                  logsNodesLong,
+		Example:               logsNodesExample,
+		ValidArgsFunction:     util.ResourceNameCompletionFunc(f, "node"),
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(f, cmd, args))
+			cmdutil.CheckErr(o.Validate())
+			cmdutil.CheckErr(o.RunLogs())
```

Comment:
This follows the old pattern, all the new commands (and eventually old ones too) will follow the pattern like in https://github.com/kubernetes/kubernetes/blob/master/staging/src//k8s.io/kubectl/pkg/cmd/events/events.go or https://github.com/kubernetes/kubernetes/blob/master/staging/src//k8s.io/kubectl/pkg/cmd/wait/wait.go where flags are separate from options struct which is meant to drop tight coupling with cobra flags.

Reply:
The `NodeLogsOptions` struct here gets converted to a `nodev1alpha1.NodeLogOptions` in the `ToNodeLogOptions()` func below. Does this simply mean the `NodeLogsOptions` struct should be renamed to `NodeLogsFlags`, or should another intermediary `NodeLogsOptions` struct be introduced here?